pkg/deploy/internal/slo/slo.go

```diff
@@ -83,6 +85,11 @@ func Deploy(ctx context.Context, client deployServiceLevelObjectiveClient, prope
 	}
 
 	//Strategy 3 is to create a new slo
+	if isSloV1, parseErr := api.IsSloV1Payload(requestPayload); parseErr != nil {
```

Comment:
We already have unmarshaled the payload at the beginning line 54. We're adding an `externalID`, with a little change we can save some time here. Also, we should validate not only for create but also update.